src/metabase/api/common.clj

```diff
@@ -413,6 +413,13 @@
     (check (not (:archived object))
       [404 {:message (tru "The object has been archived."), :error_code "archived"}])))
 
+(defn check-is-readonly
+  "Check that the object has `:is_write` = false, or throw a `405`. Returns `object` as-is if check passes."
+  [object]
+  (u/prog1 object
+    (check (not (:is_write object))
+      [405 {:message (tru "Write queries are only executable via the Actions API."), :error_code "is_not_readonly"}])))
```

Comment:
I think 405s make sense vs 400s for this but I'd double check with the FE folks that it's not going to break anything just to be sure

Reply:
Good point, I checked with them and they're not using `x == 400`.